### PR TITLE
7 post comment by article

### PR DESCRIPTION
**Pull request by Gazdean**
ticket number 7

### CODE REVIEW

File: db/models/articles-model.js [hunk 1]
```diff
@@ -16,3 +16,39 @@ exports.fetchArticlesById = (article_id) => {
...
+                status: 401, message: 'user name does not exist'
+            })
+        } else { 
+            return db.query(`
+            INSERT INTO comments(article_id, author, body)
+            VALUES ($1, $2, $3)
+            RETURNING *;`
+            , [article_id, username, body])
+            .then((result) => {
+                return result.rows;
```
Comment (B7Bprog) [inline db/models/articles-model.js]: We only need to return the object here, not the whole array. Please chenge this.

File: __tests__/api.test.js [hunk 2]
```diff
@@ -89,7 +89,62 @@ describe('GET /api/articles/:article_id', () => {
...
+    test("returns the new comment that has been created along with 201 status", () => {
+        return request(app)
+        .post("/api/articles/2/comments")
+        .send({
+            username: "butter_bridge",
+            body: "well well well"
+        })
+        .expect(201)
+        .then(({body}) => {
+            const comment = body.comment[0];
```
Comment (B7Bprog) [inline __tests__/api.test.js]: Once the model is fixed (see my other comment) comment will be an object, not an array.

File: __tests__/api.test.js [hunk 3]
```diff
@@ -89,7 +89,62 @@ describe('GET /api/articles/:article_id', () => {
...
+            const comment = body.comment[0]; 
+            expect(comment).toMatchObject({ 
+                body: "well well well",
+                votes: expect.any(Number),
+                article_id: 2,
+                author: "butter_bridge",
+                comment_id:  expect.any(Number)
+            });
+        });
+    })
```
Comment (B7Bprog) [inline __tests__/api.test.js:150]: Please add another test where the extra properties in the post object are ignored. For example when posting with this:
```
{
            username: "butter_bridge",
            body: "well well well",
            extraProp: "extraValue"
        }
```

File: __tests__/api.test.js [hunk 4]
```diff
@@ -89,7 +89,62 @@ describe('GET /api/articles/:article_id', () => {
...
+        .post('/api/articles/2/comments')
+        .send({
+        username: "butter_bridge"     
+        })
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.message).toBe('both username and comment body is required')
+        })
+    })
+    test('if user name doesnt exist in database return error code 401 and helpful message', () => {
```
Comment (B7Bprog) [inline __tests__/api.test.js]: This is supposed to be a 404 where the username is not found. (401 would be an authentication issue)

File: __tests__/api.test.js [hunk 5]
```diff
@@ -89,7 +89,62 @@ describe('GET /api/articles/:article_id', () => {
...
+        .send({
+        username: 9, 
+        body: 9
+        })
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.message).toBe('invalid data type')
+        })
+    })
+})
```
Comment (B7Bprog) [inline __tests__/api.test.js:222]: Please include two more tests here, one for when the ID is invalid, and one where there's no article with the given ID.

File: db/controllers/articles-controller.js [hunk 6]
```diff
@@ -7,4 +7,16 @@ exports.getArticlesById = (req, res, next) => {
...
-};
\ No newline at end of file
+};
+
+exports.postArticleComment = (req, res, next) => {
+    const { username, body } = req.body
+    const { article_id } = req.params
+    createArticleComment(article_id, username, body)
+    .then ((rows) => {
+        res.status(201).send({ comment: rows })
```
Comment (B7Bprog) [inline db/controllers/articles-controller.js:23]: Only an object supposed to be sent back instead of an array.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Gareth, please check my comments, refactor your code, then please send this PR back to us with the update to have a look.

File: db/models/articles-model.js (hunk 1 above)
Comment (Gazdean) [reply db/models/articles-model.js]: done

File: __tests__/api.test.js (hunk 2 above)
Comment (Gazdean) [reply __tests__/api.test.js]: done

File: __tests__/api.test.js (hunk 4 above)
Comment (Gazdean) [reply __tests__/api.test.js]: done

Comment (Gazdean) [reply __tests__/api.test.js:123]: done

File: __tests__/api.test.js (hunk 3 above)
Comment (Gazdean) [reply __tests__/api.test.js:150]: done

File: __tests__/api.test.js (hunk 5 above)
Comment (Gazdean) [reply __tests__/api.test.js:222]: done

File: db/controllers/articles-controller.js (hunk 6 above)
Comment (Gazdean) [reply db/controllers/articles-controller.js:23]: I consoled logged rows and it is an object, so i think I'm misunderstanding this

File: __tests__/api.test.js [hunk 7]
```diff
@@ -128,7 +128,98 @@ describe('GET /api/articles/:article_id', () => {
...
+            expect(comment).toMatchObject({ 
+                body: "well well well",
+                votes: expect.any(Number),
+                article_id: 2,
+                author: "butter_bridge",
+                comment_id:  expect.any(Number)
+            });
+        });
+    }) 
+    test('if user name doesnt exist in database return error code 401 and helpful message', () => {
```
Comment (pippaamy) [inline __tests__/api.test.js]: we need to change the test description  too

File: db/controllers/articles-controller.js [hunk 8]
```diff
@@ -13,4 +13,17 @@ exports.getArticlesById = (req, res, next) => {
...
-};
\ No newline at end of file
+};
+
+exports.postArticleComment = (req, res, next) => {
+    const { username, body } = req.body
+    const { article_id } = req.params
+    createArticleComment(article_id, username, body)
+    .then ((rows) => {
+        console.log(rows)
```
Comment (pippaamy) [inline db/controllers/articles-controller.js]: delete console logs before pushing up

**Review by pippaamy — COMMENTED** [review]
great work, just a couple of changes then you're all good to merge!

File: __tests__/api.test.js (hunk 7 above)
Comment (Gazdean) [reply __tests__/api.test.js]: done

File: db/controllers/articles-controller.js (hunk 8 above)
Comment (Gazdean) [reply db/controllers/articles-controller.js]: done

**Comment by Gazdean** [comment]
merged 
branch deleted

**Pull request merged**